awd/cli.py

```diff
@@ -221,87 +232,41 @@ def listen(
-            except KeyError:
-                logger.exception(
-                    "Failed to get DOI from message attributes: %s", sqs_message
+                sqs_client.process_result_message(
+                    sqs_message=sqs_message,
+                    retry_threshold=retry_threshold,
                 )
+            except (KeyError, ClientError, InvalidSQSMessageError):
+                logger.exception("Error while processing SQS message: %s", sqs_message)
                 continue
-            try:
-                body = json.loads(str(sqs_message.get("Body")))
-            except ValueError:
-                logger.exception("Failed to parse body of SQS message: %s", sqs_message)
-                continue
-            if body["ResultType"] == "error":
-                logger.exception("DOI: %s, Result: %s", doi, body)
-                sqs.delete(
-                    ctx.obj["sqs_base_url"],
-                    ctx.obj["sqs_output_queue"],
-                    sqs_message["ReceiptHandle"],
-                )
-                try:
-                    if doi_table.attempts_exceeded(doi, int(retry_threshold)):
-                        doi_table.update_status(doi, Status.FAILED.value)
-                    else:
-                        doi_table.update_status(doi, Status.UNPROCESSED.value)
-                except KeyError:
-                    logger.exception(
-                        "Key error in table while processing %s",
-                        doi,
-                    )
-                except ClientError as e:
-                    logger.exception(
-                        "Table error while processing %s: %s",
-                        doi,
-                        e.response["Error"]["Message"],
-                    )
-            else:
-                logger.info("DOI: %s, Result: %s", doi, body)
-                sqs.delete(
-                    ctx.obj["sqs_base_url"],
-                    ctx.obj["sqs_output_queue"],
-                    sqs_message["ReceiptHandle"],
-                )
-                try:
-                    doi_table.update_status(doi, Status.SUCCESS.value)
-                except KeyError:
-                    logger.exception(
-                        "Key error in table while processing %s",
-                        doi,
-                    )
-                except ClientError as e:
-                    logger.exception(
-                        "Table error while processing %s: %s",
-                        doi,
-                        e.response["Error"]["Message"],
-                    )
         logger.debug("Messages received and deleted from output queue")
     except ClientError as e:
         logger.exception(
-            "Failure while retrieving SQS messages: %s", e.response["Error"]["Message"]
+            "Error while retrieving messages from SQS queue: %s",
+            e.response["Error"]["Message"],
         )
```

Comment:
Should there be any handling if a `ClientError` is thrown?  Do you still want to continue to sending emails via SES?  Potentially just a `raise` to bubble that up to the CLI command itself?